open-company/open-company-web
# Video

Add video posts using Ziggeo sdk and out developed theme of betajs-media-components used to display the video stuff.

To test:
- click New
- enter something in the body and leave the headline blank
- [x] do you see an error message if you try to post w/o a headline? Good
- now enter a headline and click post
- [x] did it post? Good
- click New again
- click video button and start recording
- in the video talk a bit to test the transcription later
- while recording add a headline
- while still recording click post
- [x] do you see an error message related to recording still running? Good
- now stop the recording
- before picking a cover shot click post again
- [x] do you see an error message related to the missing cover image? Good
- select a cover image
- wait for the video to finish upload and processing
- [x] do you see only one notification at the end of the upload process saying that it finished uploading and processing? Good
- click Post
- [x] do you see your video in the stream view? Can you play it? Good
- [x] do you see the transcription? Good
- click New post
- add a video a headline and a body
- click Post before finishing uploading
- [x] do you see notifications of the upload progress and of the processing progress? Good
- [x] do you see the video appear in the stream view once the upload is complete? Good
- now add a text post and post it
- edit it
- add a video and save
- go back to the stream view
- [x] do you see the old post with the video and the transcription added? Good
- now edit one of the posts with a video
- remove the video and save
- [x] do you see the post w/o video and transcription in the steam? Good
- switch to grid view
- click on a post with video
- do you see the video in the fullscreen view? Good
- now go back to grid view
- share a post with video and transcription via url with all the world (not only team members)
- open the copied url in a incognito window
- [x] do you see the video player and the transcription? Good


NB: production server is using the Ziggeo sdk compiled but not loaded from the CDN, it's loaded directly from the web server. This should change, we should concat the ziggeo sdk js to oc_assets.js.
It's not a big issue only a bit slower, adding it ot oc_assets should be tested to make sure externs work.

NB: if we decide to remove transcription we need to open a PR and merge also this branch of infra https://github.com/belucid/open-company-infrastructure/tree/fix-video
